### PR TITLE
Mini Composer

### PR DESCRIPTION
**Pull request by mxstbr**
**Status**
- [ ] WIP
- [x] Ready for review
- [ ] Needs testing

**Deploy after merge (delete what needn't be deployed)**
- hyperion (frontend)

### CODE REVIEW

**Comment by mxstbr** [comment]
Reviewers note: I set up the circular dependency plugin and removed all circular dependencies because I was getting a null render as soon as I extracted the mini composer into its own component and I thought resolving all circular deps (of which we had two) would fix it.

It did not, instead it was just an incorrect import but for some reason no error was thrown 🤷‍♂️ I left it in since it seems useful to avoid adding circular dependencies which are hard to track down and could lead to behavior like that.

**Comment by brianlovin** [comment]
This is really awesome! A few notes:
- [x] can we redirect to the thread modal view after publish, instead of the full thread view?
- [x] localstorage persistence doesn't seem to be clearing the title/body inputs for me post-publish
- [x] we need a new icon for the "view full size composer" + a tooltip saying something like "Open fullscreen composer". let me see if i can find anything for the icon.
- [x] the 'Post' button becomes enabled after you've typed a title + body, even if you haven't selected a channel. It should be disabled until you have selected a channel too
- [x] the 'Post' button doesn't appear to be doing anything if I post something from the channel view
- [x] post-publish the mini composer should re-collapse. right now it just stays open any time i go back to the community/channel view

**Comment by brianlovin** [comment]
- [x] the 'Post' button is getting cut off on mobile - we'll need that bottom action bar to wrap on mobile

File: src/components/composer/inputs.js [hunk 1]
```diff
@@ -117,7 +117,7 @@ export default (props: Props) => {
                   onChange={changeTitle}
                   style={ThreadTitle}
                   value={title}
-                  placeholder={'What‘s on your mind?'}
+                  placeholder="What do you want to talk about?"
```
Comment (brianlovin) [inline src/components/composer/inputs.js:120]: I kinda liked "What's on your mind?" - why did you change this?

File: src/components/composer/inputs.js [hunk 2]
```diff
@@ -127,7 +127,7 @@ export default (props: Props) => {
                   disabled={body === null}
                   style={ThreadDescription}
                   inputRef={bodyRef}
-                  placeholder={'Add more thoughts here...'}
+                  placeholder="Elaborate here if necessary (optional)"
```
Comment (brianlovin) [inline src/components/composer/inputs.js:130]: How about "(Optional) Add more details..."

**Comment by brianlovin** [comment]
Last question: should we replace the `outsideClickHandler` with a simple cancel button? Otherwise you can get into a state where you've started typing a title, changed your mind, and now you can't close the composer. The only way to close it is to delete all draft content then click outside.

**Review by brianlovin — APPROVED** [review]
